file_io.h

```diff
@@ -1,14 +1,15 @@
 #ifndef _FAT16_H_INCLUDED
 #define _FAT16_H_INCLUDED
 
+#include <stdio.h>
 #include <unistd.h>
 #include <dirent.h>
 #include <fcntl.h>
 #include "spi.h"
 
 typedef struct
 {
-	int       fd;
+	FILE     *filp;

```

Comment:
Are you ok with replacing the file descriptor with a file stream in `fileTYPE`?  Without this change I would have to do some ugly hacks in `FileOpenEx` and you can always get the descriptor from the stream by calling `fileno(f->filp)`.